module/platform/Webots/src/Webots.cpp

```diff
@@ -292,172 +310,206 @@ namespace module::platform {
-                // Construct the ActuatorRequests message
-                ActuatorRequests actuator_requests;
-                for (auto& servo : servo_state) {
-                    if (servo.dirty) {
-                        // Servo is no longer dirty
-                        servo.dirty = false;
+            send_io = on<Every<UPDATE_FREQUENCY, Per<std::chrono::seconds>>, Single, Priority::HIGH>().then(
+                "Simulator Update Loop",
+                [this] {
+                    // Bound the time_step for the cameras and other sensors by the minimum allowed time_step for the
+                    // competition
+                    uint32_t sensor_timestep = std::max(min_sensor_time_step, time_step);
+                    uint32_t camera_timestep = std::max(min_camera_time_step, time_step);
+
+                    // Construct the ActuatorRequests message
+                    ActuatorRequests actuator_requests = create_sensor_time_steps(sensor_timestep, camera_timestep);
+                    for (auto& servo : servo_state) {
+                        if (servo.dirty) {
+                            // Servo is no longer dirty
+                            servo.dirty = false;
+
+                            // Create servo position message
+                            actuator_requests.motor_positions.emplace_back(
+                                MotorPosition(servo.name, servo.goal_position));
+
+                            // Create servo velocity message
+                            actuator_requests.motor_velocities.emplace_back(
+                                MotorVelocity(servo.name, servo.moving_speed));
+
+                            // Create servo PID message
+                            actuator_requests.motor_pids.emplace_back(
+                                MotorPID(servo.name, {servo.p_gain, servo.i_gain, servo.d_gain}));
+                        }
+                    }
 
-                        // Create servo position message
-                        actuator_requests.motor_positions.emplace_back(MotorPosition(servo.name, servo.goal_position));
+                    // Serialise ActuatorRequests
+                    std::vector<char> data =
+                        NUClear::util::serialise::Serialise<ActuatorRequests>::serialise(actuator_requests);
 
-                        // Create servo velocity message
-                        actuator_requests.motor_velocities.emplace_back(MotorVelocity(servo.name, servo.moving_speed));
+                    // Size of the message, in network endian
+                    const uint32_t Nn = htonl(data.size());
 
-                        // TODO: Add FSRs message
+                    // Send the message size first
+                    if (send(fd, &Nn, sizeof(Nn), 0) != sizeof(Nn)) {
+                        log<NUClear::ERROR>(
+                            fmt::format("Error in sending ActuatorRequests' message size,  {}", strerror(errno)));
+                    }
```

Comment:
If `connection_active` is false then we have not yet received the welcome message from the robocup controller, so we should not be sending actuator requests yet. (include the next send statement in this if statement as well)
```suggestion
                    if (!connection_active) {
                        // Send the message size first
                        if (send(fd, &Nn, sizeof(Nn), 0) != sizeof(Nn)) {
                            log<NUClear::ERROR>(
                                fmt::format("Error in sending ActuatorRequests' message size,  {}", strerror(errno)));
                        }
                    }
```